### PR TITLE
Gravity function

### PR DESCRIPTION
**Pull request by NikhilPappu**
#### Brief description of what is fixed or changed
Added a simple function that allows one to add the gravity force to bodies in an elegant way.

### CODE REVIEW

File: sympy/physics/mechanics/functions.py [hunk 1]
```diff
@@ -341,6 +341,39 @@ def potential_energy(*body):
...
+>>> po = Point('po')
+>>> pa = Particle('pa', po, m)
+>>> A = ReferenceFrame('A')
+>>> P = Point('P')
+>>> I = outer(A.x, A.x)
+>>> B = RigidBody('B', P, A, m, (I, P))
+>>> forceList = [(po, F1), (P, F2)]
+>>> forceList.extend(gravity(g*N.y, pa, B))
+>>> forcelist
+[(po, F1), (P, F2), (po, g*m*N.y), (P, g*M*N.y)]
```
Comment (moorepants) [inline sympy/physics/mechanics/functions.py]: All the above lines should be indented.

File: sympy/physics/mechanics/functions.py [hunk 2]
```diff
@@ -341,6 +341,39 @@ def potential_energy(*body):
             raise TypeError('*body must have only Particle or RigidBody')
     return pe_sys
 
+def gravity(acceleration, *body):
+    """
+    Returns a list of gravity forces given the acceleration
+    due to gravity and an iterable of bodies.
```
Comment (moorepants) [inline sympy/physics/mechanics/functions.py]: ... and an number of bodies or particles. (it doesn't take an iterable))

File: sympy/physics/mechanics/functions.py [hunk 3]
```diff
@@ -341,6 +341,39 @@ def potential_energy(*body):
             raise TypeError('*body must have only Particle or RigidBody')
     return pe_sys
 
+def gravity(acceleration, *body):
```
Comment (moorepants) [inline sympy/physics/mechanics/functions.py]: What happens if you don't pass in any bodies? Probably should error.

File: sympy/physics/mechanics/functions.py [hunk 4]
```diff
@@ -341,6 +341,39 @@ def potential_energy(*body):
...
+>>> forceList.extend(gravity(g*N.y, pa, B))
+>>> forcelist
+[(po, F1), (P, F2), (po, g*m*N.y), (P, g*M*N.y)]
+    """
+    gravity_force = []
+    for e in body:
+        if isinstance(e, Particle):
+            gravity_force.append((e.point, e.mass*acceleration))
+        elif isinstance(e, RigidBody):
+            gravity_force.append((e.masscenter, e.mass*acceleration))
```
Comment (moorepants) [inline sympy/physics/mechanics/functions.py]: Another way to do this that is, maybe, more Pythonic:

```
try:
    point = e.masscenter
except AttributeError:
    point = e.point
```

File: sympy/physics/mechanics/functions.py (hunk 3 above)
Comment (moorepants) [reply sympy/physics/mechanics/functions.py]: I'd call it "bodies" instead of "body".

**Review by moorepants — CHANGES_REQUESTED** [review]
This is a long needed convenience function. Thanks for adding it. Please address my comments and we can add this.

**Comment by NikhilPappu** [comment]
@moorepants Travis isn't passing it. Its showing errors unrelated to this
`________________________________________________________________________________
________________________ sympy.codegen.cfunctions.log1p ________________________
File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/sympy/codegen/cfunctions.py", line 100, in sympy.codegen.cfunctions.log1p
Failed example:
    '%.0e' % log1p(1e-99).evalf()
Expected:
    '1e-99'
Got:
    '0e+00'
________________________________________________________________________________
_____________________ sympy.solvers.solveset.substitution ______________________
File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/sympy/solvers/solveset.py", line 1548, in sympy.solvers.solveset.substitution
Failed example:
    substitution(eqs, [y, z])
Expected:
    {(-log(3), -sqrt(-exp(2*x) - sin(log(3)))),
    (-log(3), sqrt(-exp(2*x) - sin(log(3)))),
    (ImageSet(Lambda(_n, 2*_n*I*pi + Mod(-log(3), 2*I*pi)), S.Integers),
    ImageSet(Lambda(_n, -sqrt(-exp(2*x) + sin(2*_n*I*pi +
    Mod(-log(3), 2*I*pi)))), S.Integers)),
    (ImageSet(Lambda(_n, 2*_n*I*pi + Mod(-log(3), 2*I*pi)), S.Integers),
    ImageSet(Lambda(_n, sqrt(-exp(2*x) + sin(2*_n*I*pi +
        Mod(-log(3), 2*I*pi)))), S.Integers))}
Got:
    {(log(1/3), -sqrt(-exp(2*x) + sin(log(1/3)))), (log(1/3), sqrt(-exp(2*x) + sin(log(1/3)))), (log(1/3), -sqrt(-exp(2*x) - sin(log(3)))), (log(1/3), sqrt(-exp(2*x) - sin(log(3)))), (ImageSet(Lambda(_n, 2*_n*I*pi + Mod(log(1/3), 2*I*pi)), S.Integers), ImageSet(Lambda(_n, -sqrt(-exp(2*x) + sin(2*_n*I*pi + Mod(-log(3), 2*I*pi)))), S.Integers)), (ImageSet(Lambda(_n, 2*_n*I*pi + Mod(log(1/3), 2*I*pi)), S.Integers), ImageSet(Lambda(_n, sqrt(-exp(2*x) + sin(2*_n*I*pi + Mod(-log(3), 2*I*pi)))), S.Integers))}
`

File: sympy/physics/mechanics/functions.py [hunk 5]
```diff
@@ -341,6 +341,45 @@ def potential_energy(*body):
...
 
+def gravity(acceleration, *bodies):
+    """
+    Returns a list of gravity forces given the acceleration
+    due to gravity and any number of particles or rigidbodies.
+
+    Examples
+    ========
+
+>>> from sympy.physics.mechanics import ReferenceFrame, Point, Particle, outer, RigidBody
```
Comment (asmeurer) [inline sympy/physics/mechanics/functions.py]: The full docstring should be indented four spaces.

Comment (NikhilPappu) [reply sympy/physics/mechanics/functions.py]: Sorry about that. I am not able to run bin/doctest for some reason. I am able to run bin/test.
This is the error I am getting 

Traceback (most recent call last):
  File "C:\Python36\lib\site-packages\theano\gof\lazylinker_c.py", line 75, in <module>
    raise ImportError()
ImportError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python36\lib\site-packages\theano\gof\lazylinker_c.py", line 92, in <module>
    raise ImportError()
ImportError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\pappu\Desktop\Diving_Into_SymPy\sympy4\sympy\sympy\utilities\runtests.py", line 712, in _doctest
    if import_module('theano') is None:
  File "C:\Users\pappu\Desktop\Diving_Into_SymPy\sympy4\sympy\sympy\external\importtools.py", line 133, in import_module
    mod = __import__(module, **__import__kwargs)
  File "C:\Python36\lib\site-packages\theano\__init__.py", line 110, in <module>
    from theano.compile import (
  File "C:\Python36\lib\site-packages\theano\compile\__init__.py", line 12, in <module>
    from theano.compile.mode import *
  File "C:\Python36\lib\site-packages\theano\compile\mode.py", line 11, in <module>
    import theano.gof.vm
  File "C:\Python36\lib\site-packages\theano\gof\vm.py", line 673, in <module>
    from . import lazylinker_c
  File "C:\Python36\lib\site-packages\theano\gof\lazylinker_c.py", line 127, in <module>
    preargs=args)
  File "C:\Python36\lib\site-packages\theano\gof\cmodule.py", line 2359, in compile_str
    (status, compile_stderr.replace('\n', '. ')))
Exception: Compilation failed (return status=1): In file included from C:/mingw/.                  from C:\Users\pappu\AppData\Local\Theano\compiledir_Windows-10-10.0.15063-SP0-Intel64_Family_6_Model_78_Stepping_3_GenuineIntel-3.6.4-32\lazy. C:/mingw/mingw32/lib/gcc/i686-w64-mingw32/7.1.0/include/c++/cmath:1136:11: err.            ^~~~~;ot been declared

You can find the C code in this temporary file: C:\Users\pappu\AppData\Local\Temp\theano_compilation_error_ewxf8ay5

**Comment by NikhilPappu** [comment]
@moorepants I made the requested changes. Is it good enough to be pushed now?

**Review by moorepants — APPROVED** [review]

**Comment by moorepants** [comment]
Looks good! Thanks.

**Pull request merged**